Client/Frontend/Browser/TabManager.swift

```diff
@@ -193,7 +193,10 @@ class TabManager: NSObject {
         if previous === tab {
             return
         }
-
+        // Convert the global mode to private if opening private tab from normal tab/ history/bookmark.
+        if selectedTab?.isPrivate == false && tab?.isPrivate == true {
+            PrivateBrowsingManager.shared.isPrivateBrowsing = true
```

Comment:
Does this ever get turned back off?

Reply:
When in private mode the only way you can go back to normal is by tab tray. When you switch mode from tab tray it resets this value.

Reply:
Oh I see, I thought previously you said this wasn't used anywhere, but just not in all instances of getting into private mode.